RecoLocalTracker/SiPixelRecHits/src/PixelCPETemplateReco.cc

```diff
@@ -254,20 +254,82 @@ LocalPoint PixelCPETemplateReco::localPosition(DetParam const& theDetParam, Clus
   // ******************************************************************
 
   // Check exit status
-  if
-    UNLIKELY(theClusterParam.ierr != 0) {
+  if UNLIKELY (theClusterParam.ierr != 0) {
+    LogDebug("PixelCPETemplateReco::localPosition")
+        << "reconstruction failed with error " << theClusterParam.ierr << "\n";
+
+    // Gavril: what do we do in this case ? For now, just return the cluster center of gravity in microns
+    // In the x case, apply a rough Lorentz drift average correction
+    // To do: call PixelCPEGeneric whenever PixelTempReco1D fails
+    float lorentz_drift = -999.9;
+    if (!fpix)
+      lorentz_drift = 60.0f;  // in microns
+    else
+      lorentz_drift = 10.0f;  // in microns
+    // ggiurgiu@jhu.edu, 21/09/2010 : trk angles needed to correct for bows/kinks
+    if (theClusterParam.with_track_angle) {
+      theClusterParam.templXrec_ =
+          theDetParam.theTopol->localX(theClusterParam.theCluster->x(), theClusterParam.loc_trk_pred) -
+          lorentz_drift * micronsToCm;  // rough Lorentz drift correction
+      theClusterParam.templYrec_ =
+          theDetParam.theTopol->localY(theClusterParam.theCluster->y(), theClusterParam.loc_trk_pred);
+    } else {
+      edm::LogError("PixelCPETemplateReco") << "@SUB = PixelCPETemplateReco::localPosition"
+                                            << "Should never be here. PixelCPETemplateReco should always be called "
+                                               "with track angles. This is a bad error !!! ";
+
+      theClusterParam.templXrec_ = theDetParam.theTopol->localX(theClusterParam.theCluster->x()) -
+                                   lorentz_drift * micronsToCm;  // rough Lorentz drift correction
+      theClusterParam.templYrec_ = theDetParam.theTopol->localY(theClusterParam.theCluster->y());
+    }
+  } else if UNLIKELY (UseClusterSplitter_ && theClusterParam.templQbin_ == 0) {
+    cout << " PixelCPETemplateReco : We should never be here !!!!!!!!!!!!!!!!!!!!!!" << endl;
```

Comment:
please remove cout, this could be replaced with edm::LogError or abort the program with exception or assert, as is more suitable.